diff --git a/tensorflow/tsl/platform/default/build_config/BUILD b/tensorflow/tsl/platform/default/build_config/BUILD
index 088f4c4e752..b249fcb0d59 100644
--- a/tensorflow/tsl/platform/default/build_config/BUILD
+++ b/tensorflow/tsl/platform/default/build_config/BUILD
@@ -111,6 +111,9 @@ cc_library(
         "//conditions:default": [
             "-Wl,-rpath,../local_config_cuda/cuda/lib64",
             "-Wl,-rpath,../local_config_cuda/cuda/extras/CUPTI/lib64",
+            "-Wl,-rpath,${PREFIX}/lib",
+            "-Wl,-rpath-link,${PREFIX}/lib/stubs",
+            "-L${PREFIX}/lib",
         ],
     }),
     deps = [
