From a1af44e65ba26ee3d0d7d4087f3330cbf9cd82c1 Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Wed, 27 Jun 2018 12:46:18 -0500
Subject: [PATCH 01/13] add rt to linkopts

---
 tensorflow/BUILD          | 1 +
 tensorflow/tensorflow.bzl | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index b5e0a4e..b16cc44 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -499,6 +499,7 @@ tf_cc_shared_object(
         "//tensorflow:darwin": [],
         "//tensorflow:windows": [],
         "//conditions:default": [
+            "-lrt",
             "-Wl,--version-script",  #  This line must be directly followed by the version_script.lds file
             "$(location //tensorflow:tf_framework_version_script.lds)",
         ],
diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index adac895..de13f1c 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -355,7 +355,7 @@ def tf_cc_shared_object(
         srcs = [],
         deps = [],
         data = [],
-        linkopts = [],
+        linkopts = ['-lrt'],
         framework_so = tf_binary_additional_srcs(),
         kernels = [],
         **kwargs):
@@ -391,7 +391,7 @@ def tf_cc_binary(
         srcs = [],
         deps = [],
         data = [],
-        linkopts = [],
+        linkopts = ['-lrt'],
         copts = tf_copts(),
         kernels = [],
         **kwargs):
-- 
2.7.4

