From efc3b84c1b75e0e3af40d5cafb871d4c1371c8bd Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Thu, 4 Apr 2019 07:52:48 -0500
Subject: [PATCH 13/13] hardcode internal nccl version in configure

---
 configure.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.py b/configure.py
index 4f8cae2..acae56e 100644
--- a/configure.py
+++ b/configure.py
@@ -1116,6 +1116,9 @@ def set_tf_nccl_install_path(environ_cp):
       '[Default is to use https://github.com/nvidia/nccl]: ')
 
   for _ in range(_DEFAULT_PROMPT_ASK_ATTEMPTS):
+    # force use of included open source version of nccl
+    tf_nccl_version = ''
+    break
     tf_nccl_version = get_from_env_or_user_or_default(
         environ_cp, 'TF_NCCL_VERSION', ask_nccl_version, '')
 
-- 
2.7.4

