From b45c949ec75099034f76e044f31a9dceab84e288 Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Wed, 27 Jun 2018 12:46:18 -0500
Subject: [PATCH] add rt to linkopts

---
 tensorflow/tensorflow.bzl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index b59f8e1f98..1db5f9eedd 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -296,7 +296,7 @@ def tf_cc_shared_object(
     name,
     srcs=[],
     deps=[],
-    linkopts=[],
+    linkopts=['-lrt'],
     framework_so=tf_binary_additional_srcs(),
     **kwargs):
   native.cc_binary(
@@ -327,7 +327,7 @@ register_extension_info(
 def tf_cc_binary(name,
                  srcs=[],
                  deps=[],
-                 linkopts=[],
+                 linkopts=['-lrt'],
                  copts=tf_copts(),
                  **kwargs):
   native.cc_binary(
-- 
2.11.1

