diff -urN a/third_party/curl.BUILD b/third_party/curl.BUILD
--- a/third_party/curl.BUILD	2022-05-23 01:24:29.000000000 +0300
+++ b/third_party/curl.BUILD	2022-07-26 12:34:28.980437660 +0300
@@ -572,7 +572,6 @@
         "#  define OS \"x86_64-apple-darwin15.5.0\"",
         "#  define USE_SECTRANSP 1",
         "#else",
-        "#  define CURL_CA_BUNDLE \"/etc/ssl/certs/ca-certificates.crt\"",
         "#  define GETSERVBYPORT_R_ARGS 6",
         "#  define GETSERVBYPORT_R_BUFSIZE 4096",
         "#  define HAVE_BORINGSSL 1",
