From d1f335d089cf2a5465cc5111d59d02a59612ef9f Mon Sep 17 00:00:00 2001
From: Jonathan Helmus <jjhelmus@gmail.com>
Date: Wed, 27 Jun 2018 12:46:18 -0500
Subject: [PATCH 1/7] add rt to linkopts

---
 tensorflow/BUILD          | 1 +
 tensorflow/tensorflow.bzl | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index a04ddf9f8a..fb0596bb5d 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -498,6 +498,7 @@ tf_cc_shared_object(
         "//tensorflow:macos": [],
         "//tensorflow:windows": [],
         "//conditions:default": [
+            "-lrt",
             "-Wl,--version-script,$(location //tensorflow:tf_framework_version_script.lds)",
         ],
     }),
diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index 98e2cf100b..1d4ce39599 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -473,7 +473,7 @@ def tf_cc_shared_object(
         srcs = [],
         deps = [],
         data = [],
-        linkopts = [],
+        linkopts = ['-lrt'],
         framework_so = tf_binary_additional_srcs(),
         soversion = None,
         kernels = [],
@@ -577,7 +577,7 @@ def tf_cc_binary(
         srcs = [],
         deps = [],
         data = [],
-        linkopts = [],
+        linkopts = ['-lrt'],
         copts = tf_copts(),
         kernels = [],
         per_os_targets = False,  # Generate targets with SHARED_LIBRARY_NAME_PATTERNS
-- 
2.17.1

