From 230d31c5a4b0df20c916cd3cbbe85be4d7b5cf1b Mon Sep 17 00:00:00 2001
From: Forrest Waters <fwaters@0609.lan>
Date: Wed, 31 Jul 2019 11:35:03 -0500
Subject: [PATCH] disable prompt for ios support

---
 configure.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.py b/configure.py
index 43af22de..70ab3d26 100644
--- a/configure.py
+++ b/configure.py
@@ -1541,9 +1541,9 @@ def main():
 
   system_specific_test_config(os.environ)
 
-  set_action_env_var(environ_cp, 'TF_CONFIGURE_IOS', 'iOS', False)
-  if environ_cp.get('TF_CONFIGURE_IOS') == '1':
-    configure_ios()
+  #set_action_env_var(environ_cp, 'TF_CONFIGURE_IOS', 'iOS', False)
+  #if environ_cp.get('TF_CONFIGURE_IOS') == '1':
+  #  configure_ios()
 
   print('Preconfigured Bazel build configs. You can use any of the below by '
         'adding "--config=<>" to your build command. See .bazelrc for more '
-- 
2.20.1 (Apple Git-117)

