From 0a1faf0711ad275111dcb0a031904c0acd68a2f7 Mon Sep 17 00:00:00 2001
From: PowerAI CI Functional ID <pwraiciw@us.ibm.com>
Date: Thu, 7 May 2020 09:46:54 +0000
Subject: [PATCH] SQlite BUILD changes for external sqlite package

---
 third_party/systemlibs/sqlite.BUILD | 29 ++++++++++++++++++++++++++++-
 1 file changed, 28 insertions(+), 1 deletion(-)

diff --git a/third_party/systemlibs/sqlite.BUILD b/third_party/systemlibs/sqlite.BUILD
index 20ee1eb..4ac172e 100644
--- a/third_party/systemlibs/sqlite.BUILD
+++ b/third_party/systemlibs/sqlite.BUILD
@@ -1,12 +1,39 @@
 licenses(["unencumbered"])  # Public Domain
 
+HEADERS = [
+   "sqlite3.h",
+   "sqlite3ext.h",
+]
+
+LIBS = [
+   "libsqlite3.so",
+   "libsqlite3.so.0",
+   "libsqlite3.so.0.8.6",
+   "libsqlite3.a",
+]
+
 # Production build of SQLite library that's baked into TensorFlow.
 cc_library(
     name = "org_sqlite",
-    linkopts = ["-lsqlite3"],
+    hdrs = HEADERS,
+    srcs = LIBS,
+    includes = ["."],
     visibility = ["//visibility:public"],
 )
 
+genrule(
+    name = "sqlite-files",
+    outs = HEADERS + LIBS,
+    cmd = """
+      cp -fL "$(INCLUDEDIR)/sqlite3.h" "$(@D)" &&
+      cp -fL "$(INCLUDEDIR)/sqlite3ext.h" "$(@D)" &&
+      cp -fL "$(LIBDIR)/libsqlite3.so.0.8.6" "$(@D)" &&
+      ln -sf "$(LIBDIR)/libsqlite3.so.0.8.6" "$(@D)/libsqlite3.so.0" &&
+      ln -sf "$(LIBDIR)/libsqlite3.so.0.8.6" "$(@D)/libsqlite3.so" &&
+      cp -fL "$(LIBDIR)/libsqlite3.a" "$(@D)"
+    """,
+)
+
 # This is a Copybara sync helper for Google.
 py_library(
     name = "python",
-- 
1.8.3.1

