Index: work/tensorflow/tools/pip_package/setup.py
===================================================================
--- work.orig/tensorflow/tools/pip_package/setup.py
+++ work/tensorflow/tools/pip_package/setup.py
@@ -78,21 +78,21 @@ REQUIRED_PACKAGES = [
     # NOTE: As numpy has releases that break semver guarantees and several other
     # deps depend on numpy without an upper bound, we must install numpy before
     # everything else.
-    'numpy ~= 1.19.2',
+    'numpy >= 1.19',
     # Install other dependencies
     'absl-py ~= 0.10',
     'astunparse ~= 1.6.3',
-    'flatbuffers ~= 1.12.0',
+    'flatbuffers ~= 1.12',
     'google_pasta ~= 0.2',
-    'h5py ~= 3.1.0',
-    'keras_preprocessing ~= 1.1.2',
+    'h5py >= 3.1.0',
+    'keras_preprocessing >= 1.1.2',
     'opt_einsum ~= 3.3.0',
     'protobuf >= 3.9.2',
-    'six ~= 1.15.0',
-    'termcolor ~= 1.1.0',
-    'typing_extensions ~= 3.7.4',
-    'wheel ~= 0.35',
-    'wrapt ~= 1.12.1',
+    'six >= 1.15.0',
+    'termcolor >= 1.1.0',
+    'typing_extensions >= 3.7.4',
+    'wheel >= 0.35',
+    'wrapt >= 1.11.2',
     # These packages need to be pinned exactly as newer versions are
     # incompatible with the rest of the ecosystem
     'gast == 0.4.0',
@@ -104,7 +104,7 @@ REQUIRED_PACKAGES = [
     'tensorflow-estimator >= 2.5.0rc0 , < 2.6.0',
     # TODO(scottzhu): OSS keras hasn't been formally released yet.
     # Use keras-nightly at the moment.
-    'keras-nightly ~= 2.5.0.dev',
+    'keras >= 2.4.0',
 ]
 
 
