diff --git a/tensorflow/core/platform/default/build_config/BUILD b/tensorflow/core/platform/default/build_config/BUILD
index 44a89c3a96..6d7a6f4b63 100644
--- a/tensorflow/core/platform/default/build_config/BUILD
+++ b/tensorflow/core/platform/default/build_config/BUILD
@@ -204,6 +204,9 @@ cc_library(
         "//conditions:default": [
             "-Wl,-rpath,../local_config_cuda/cuda/lib64",
             "-Wl,-rpath,../local_config_cuda/cuda/extras/CUPTI/lib64",
+            "-Wl,-rpath,/usr/local/cuda/lib64",
+            "-Wl,-rpath,/usr/local/cuda/lib64/stubs",
+            "-L${PREFIX}/lib",
         ],
     }),
     deps = [
